### ISSUE TITLE
python 3.5.2 : `pip install mysqlclient` error

### ISSUE THREAD

**Issue by zwl1619** [PyMySQL/mysqlclient-python#99]
My OS is `win 10 x64`, python is `3.5.2`,   
I am installing `mysqlclient` via `pip install mysqlclient`,   
there are some errors,as follow:

```
    D:\demo>pip install mysqlclient
    Collecting mysqlclient
      Using cached mysqlclient-1.3.7.tar.gz
    Installing collected packages: mysqlclient
      Running setup.py install for mysqlclient ... error
        Complete output from command c:\users\dell\appdata\local\programs\python\python35-32\python.exe -u -c "import setuptools, tokenize;__file__='C:\\Users\\dell\\AppData\\Local\\Temp\\pip-build-95hytmeq\\mysqlclient\\setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record C:\Users\dell\AppData\Local\Temp\pip-wv5bnfnn-record\install-record.txt --single-version-externally-managed --compile:
        running install
        running build
        running build_py
        creating build
        creating build\lib.win32-3.5
        copying _mysql_exceptions.py -> build\lib.win32-3.5
        creating build\lib.win32-3.5\MySQLdb
        copying MySQLdb\__init__.py -> build\lib.win32-3.5\MySQLdb
        copying MySQLdb\compat.py -> build\lib.win32-3.5\MySQLdb
        copying MySQLdb\converters.py -> build\lib.win32-3.5\MySQLdb
        copying MySQLdb\connections.py -> build\lib.win32-3.5\MySQLdb
        copying MySQLdb\cursors.py -> build\lib.win32-3.5\MySQLdb
        copying MySQLdb\release.py -> build\lib.win32-3.5\MySQLdb
        copying MySQLdb\times.py -> build\lib.win32-3.5\MySQLdb
        creating build\lib.win32-3.5\MySQLdb\constants
        copying MySQLdb\constants\__init__.py -> build\lib.win32-3.5\MySQLdb\constants
        copying MySQLdb\constants\CR.py -> build\lib.win32-3.5\MySQLdb\constants
        copying MySQLdb\constants\FIELD_TYPE.py -> build\lib.win32-3.5\MySQLdb\constants
        copying MySQLdb\constants\ER.py -> build\lib.win32-3.5\MySQLdb\constants
        copying MySQLdb\constants\FLAG.py -> build\lib.win32-3.5\MySQLdb\constants
        copying MySQLdb\constants\REFRESH.py -> build\lib.win32-3.5\MySQLdb\constants
        copying MySQLdb\constants\CLIENT.py -> build\lib.win32-3.5\MySQLdb\constants
        running build_ext
        building '_mysql' extension
        creating build\temp.win32-3.5
        creating build\temp.win32-3.5\Release
        C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -Dversion_info=(1,3,7,'final',1) -D__version__=1.3.7 "-IC:\Program Files (x86)\MySQL\MySQL Connector C 6.1\include" -Ic:\users\dell\appdata\local\programs\python\python35-32\include -Ic:\users\dell\appdata\local\programs\python\python35-32\include "-IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.10240.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\8.1\include\shared" "-IC:\Program Files (x86)\Windows Kits\8.1\include\um" "-IC:\Program Files (x86)\Windows Kits\8.1\include\winrt" /Tc_mysql.c /Fobuild\temp.win32-3.5\Release\_mysql.obj /Zl
        _mysql.c
        _mysql.c(29): fatal error C1083: Cannot open include file: 'my_config.h': No such file or directory
        error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\BIN\\cl.exe' failed with exit status 2
    
        ----------------------------------------
    Command "c:\users\dell\appdata\local\programs\python\python35-32\python.exe -u -c "import setuptools, tokenize;__file__='C:\\Users\\dell\\AppData\\Local\\Temp\\pip-build-95hytmeq\\mysqlclient\\setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record C:\Users\dell\AppData\Local\Temp\pip-wv5bnfnn-record\install-record.txt --single-version-externally-managed --compile" failed with error code 1 in C:\Users\dell\AppData\Local\Temp\pip-build-95hytmeq\mysqlclient\
```

What should I do?

**Comment by methane** [comment]
> What should I do?

Mail to Oracle to release new MySQL Connector/C which support VC14 runtime.
Python 3.5 is built with VC14, but new MySQL Connector/C has not released for a very long time.

Duplicate of #54

**Comment by zwl1619** [comment]
@methane 
If I change `mysql` to `mariadb`,is `MySQL Connector/C`  also necessary?

**Comment by methane** [comment]
Yes.  MariaDB uses same protocol as MySQL.
But MariaDB also releasing MariaDB Connector/C which is almost compatible with MySQL Connector/C.
When I tried MariaDB Connector/C, I can't build by some reason I forgot.
I'll try again it.

**Comment by methane** [comment]
I've failed to build.
MariaDB doesn't support VC14 too, maybe.

**Comment by topperwithharley** [comment]
Dear all, I am having the exact same problem. After mastering a marathon of reading forum posts and potential explanations why nobody is able to solve issues realted to using python in with MySQL I want to put a singular / easy question to you: 

Is there any way using Windows 10 to connect python (version 3.x.x 64 or 32 bit) to MySQL (Version 5.x.xx  64 bit / Maria DB 64 bit) to retrieve some data?

I tried to pip install the wheels available here: https://pypi.python.org/pypi/mysqlclient/1.3.7 on Python 3.5.2 32 and 64 bit as well as on Python 3.4 64 bit. I allways received following error message:

xxx.whl is not a supported wheel on this platform.

I think I am just reviving recent posts. Sorry for that already, but I just want to be sure I didn't miss anything and the misery of python - mysql is still real...

**Comment by methane** [comment]
You can see there are wheel or not on PyPI page.
Currently, wheel is available for Python 3.4 + mysqlclient 1.3.6.
If you want newer wheel, please post a comment on MySQL forum, not here.
